fontir/src/variations.rs

```diff
@@ -0,0 +1,795 @@
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{self}"))
+    }
+}
+
+impl Display for Tent {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let comment = if self.validate() { "" } else { " (invalid)" };
+        f.write_fmt(format_args!(
+            "Tent {{{}, {}, {}{}}}",
+            self.lower.into_inner(),
+            self.peak.into_inner(),
+            self.upper.into_inner(),
+            comment
+        ))?;
+        Ok(())
+    }
+}
+
+/// Split space into regions.
+///
+/// VariationModel::_locationsToRegions in Python.
+/// <https://github.com/fonttools/fonttools/blob/2f1f5e5e7be331d960a0e30d537c2b4c70d89285/Lib/fontTools/varLib/models.py#L416>
+fn regions_for(locations: &[NormalizedLocation]) -> Vec<VariationRegion> {
+    let mut minmax = HashMap::<&String, (NormalizedCoord, NormalizedCoord)>::new();
+    for location in locations.iter() {
+        for (axis, value) in location.iter() {
+            let (min, max) = minmax
+                .entry(axis)
+                .or_insert_with(|| (NormalizedCoord::new(0.0), NormalizedCoord::new(0.0)));
+            if value < min {
+                *min = *value;
+            }
+            if value > max {
+                *max = *value;
+            }
+        }
+    }
+
+    locations
+        .iter()
+        .map(|location| {
+            let mut region = VariationRegion::new();
+            for (axis, value) in location.iter() {
+                // Python just scrubs 0's out of the location's. We elect to store representative tents.
+                let (min, max) = if value.into_inner() == ZERO {
+                    (NormalizedCoord::new(ZERO), NormalizedCoord::new(ZERO))
+                } else {
+                    *minmax.get(axis).unwrap()
+                };
+                region.0.insert(axis.clone(), Tent::new(min, *value, max));
+            }
+            region
+        })
+        .collect()
+}
+
+/// Compute the influence of each master, if any, on each region.
+///
+/// The regions must have been sorted by the [LocationSortingHat] as this ensures
+/// that for each region we only need to look at preceeding regions for overlaps.
+///
+/// VariationModel::_computeMasterSupports in Python.
+/// <https://github.com/fonttools/fonttools/blob/2f1f5e5e7be331d960a0e30d537c2b4c70d89285/Lib/fontTools/varLib/models.py#L360>
+fn master_influence(regions: Vec<VariationRegion>) -> Vec<VariationRegion> {
+    let mut influence = Vec::new();
+    for (i, region) in regions.iter().enumerate() {
+        let axes: HashSet<&String> = region.0.keys().collect();
+        let mut region = region.clone();
+        for prev_region in regions[..i].iter() {
+            if axes != prev_region.0.keys().collect() {
+                continue;
+            }
+            // If prev doesn't overlap current we aren't interested
+            let overlap = region.0.iter().all(|(axis, axis_region)| {
```

Comment:
```diff
- let overlap = region.0.iter().all(|(axis, axis_region)| {
+ let overlap = region.0.iter().all(|(axis, axis_tent)| {
```

I would reserve the term "region" for the multi-axis `Dict[AxisTag, Tuple[float, float, float]]` only, and prefer "tent" for the latter (lower, peak, upper) triplet that's associated with a given axis.
On the other hand, one 'tent' could still be seen as a 'region' with only one axis..

Reply:
Ew good catch, that's a terrible terrible naming choice. I updated it to axis_name, tent.